File: app/src/main/java/com/zerostress/manager/ScheduleActivity.java
package com.zerostress.manager;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.FirebaseFirestore;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import java.util.Locale;

public class ScheduleActivity extends AppCompatActivity {

    private RecyclerView rvSchedules;
    private TextView tvEmpty;
    private FirebaseFirestore db;
    private FirebaseAuth auth;
    private List<DocumentSnapshot> schedules = new ArrayList<>();
    private boolean isAdmin = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_schedule);

        db = FirebaseFirestore.getInstance();
        auth = FirebaseAuth.getInstance();
        rvSchedules = findViewById(R.id.rvSchedules);
        tvEmpty = findViewById(R.id.tvEmpty);

        rvSchedules.setLayoutManager(new LinearLayoutManager(this));
        checkUserRole();
        loadSchedules();
    }

    private void checkUserRole() {
        String uid = auth.getUid();
        if (uid == null) return;
        db.collection("players").document(uid).get()
            .addOnSuccessListener(doc -> {
                if (doc.exists()) {
                    isAdmin = "admin".equals(doc.getString("role"));
                }
            });
    }

    private void loadSchedules() {
        db.collection("match_schedules").orderBy("matchTime").limit(30)
            .addSnapshotListener((snapshots, e) -> {
                if (e != null || snapshots == null) return;
                schedules.clear();
                for (DocumentSnapshot doc : snapshots.getDocuments()) {
                    schedules.add(doc);
                }
                if (schedules.isEmpty()) {
                    tvEmpty.setVisibility(View.VISIBLE);
                    rvSchedules.setVisibility(View.GONE);
                } else {
                    tvEmpty.setVisibility(View.GONE);
                    rvSchedules.setVisibility(View.VISIBLE);
                    rvSchedules.setAdapter(new ScheduleAdapter());
                }
            });
    }

    private void deleteSchedule(String docId) {
        new AlertDialog.Builder(this)
            .setTitle("🗑️ Delete Schedule")
            .setMessage("Are you sure you want to delete this schedule?")
            .setPositiveButton("Delete", (d, w) -> {
                db.collection("match_schedules").document(docId).delete()
                    .addOnSuccessListener(v -> {
                        Toast.makeText(this, "Schedule deleted!", Toast.LENGTH_SHORT).show();
                    })
                    .addOnFailureListener(e -> {
                        Toast.makeText(this, "Error: " + e.getMessage(), Toast.LENGTH_SHORT).show();
                    });
            })
            .setNegativeButton("Cancel", null)
            .show();
    }

    class ScheduleAdapter extends RecyclerView.Adapter<ScheduleAdapter.VH> {
        @Override
        public VH onCreateViewHolder(ViewGroup parent, int viewType) {
            View v = getLayoutInflater().inflate(R.layout.item_schedule, parent, false);
            return new VH(v);
        }

        @Override
        public void onBindViewHolder(VH holder, int position) {
            DocumentSnapshot doc = schedules.get(position);
            holder.tvTitle.setText(doc.getString("title"));

            String desc = doc.getString("description");
            String type = doc.getString("type");
            if (desc != null && !desc.isEmpty()) {
                holder.tvDesc.setText(desc);
            } else if (type != null && !type.isEmpty()) {
                holder.tvDesc.setText(type);
            } else {
                holder.tvDesc.setText("No description");
            }

            // Show user-entered dateTime if available, else format matchTime
            String userDateTime = doc.getString("dateTime");
            if (userDateTime != null && !userDateTime.isEmpty() && !userDateTime.equals("TBD")) {
                holder.tvTime.setText("🕐 " + userDateTime);
            } else {
                Long time = doc.getLong("matchTime");
                if (time != null) {
                    SimpleDateFormat sdf = new SimpleDateFormat("MMM dd, yyyy HH:mm", Locale.getDefault());
                    holder.tvTime.setText("🕐 " + sdf.format(new Date(time)));
                }
            }

            String status = doc.getString("status");
            holder.tvStatus.setText(status != null ? status : "Upcoming");

            // Show delete button for admin
            if (isAdmin) {
                holder.btnDelete.setVisibility(View.VISIBLE);
                holder.btnDelete.setOnClickListener(v -> deleteSchedule(doc.getId()));
            } else {
                holder.btnDelete.setVisibility(View.GONE);
            }
        }

        @Override
        public int getItemCount() { return schedules.size(); }

        class VH extends RecyclerView.ViewHolder {
            TextView tvTitle, tvDesc, tvTime, tvStatus;
            ImageButton btnDelete;
            VH(View v) {
                super(v);
                tvTitle = v.findViewById(R.id.tvScheduleTitle);
                tvDesc = v.findViewById(R.id.tvScheduleDesc);
                tvTime = v.findViewById(R.id.tvScheduleTime);
                tvStatus = v.findViewById(R.id.tvScheduleStatus);
                btnDelete = v.findViewById(R.id.btnDeleteSchedule);
            }
        }
    }
}

File: app/src/main/java/com/zerostress/manager/ScheduleActivity.kt
package com.zerostress.manager

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.google.firebase.firestore.DocumentSnapshot
import com.google.firebase.firestore.FirebaseFirestore
import com.zerostress.manager.ui.theme.*
import java.text.SimpleDateFormat
import java.util.*

class ScheduleActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent { ZeroStressTheme { ScheduleScreen() } }
    }
}

@Composable
fun ScheduleScreen() {
    var schedules by remember { mutableStateOf(listOf<DocumentSnapshot>()) }

    LaunchedEffect(Unit) {
        FirebaseFirestore.getInstance().collection("match_schedules").addSnapshotListener { s, _ ->
            if (s != null) schedules = s.documents
        }
    }

    Column(modifier = Modifier.fillMaxSize().padding(16.dp)) {
        Text("📅 Match Schedule", fontSize = 22.sp, fontWeight = FontWeight.Bold, color = Color.White)
        Spacer(modifier = Modifier.height(12.dp))

        if (schedules.isEmpty()) {
            Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
                Text("No scheduled matches", color = ZSTextMuted)
            }
        } else {
            LazyColumn(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                items(schedules) { doc ->
                    val title = doc.getString("title") ?: "Match"
                    val dateTime = doc.getString("dateTime") ?: "TBD"
                    val type = doc.getString("type") ?: "Custom"
                    val status = doc.getString("status") ?: "Upcoming"

                    Card(modifier = Modifier.fillMaxWidth(), shape = RoundedCornerShape(12.dp), colors = CardDefaults.cardColors(containerColor = ZSCard)) {
                        Column(modifier = Modifier.padding(14.dp)) {
                            Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
                                Text(title, color = Color.White, fontWeight = FontWeight.Bold, fontSize = 16.sp)
                                Text(status, color = ZSPrimary, fontSize = 12.sp)
                            }
                            Text("🕐 $dateTime", color = ZSTextMuted, fontSize = 13.sp)
                            Text("Type: $type", color = ZSTextMuted, fontSize = 13.sp)
                        }
                    }
                }
            }
        }
    }
}
